torch/fx/experimental/rewriter.py

```diff
@@ -0,0 +1,64 @@
+import ast
+import astor
+import inspect
+import textwrap
+import copy
+from types import FunctionType
+
+class AST_Rewriter(ast.NodeTransformer):
+    '''
+    Take a FunctionType object representing a `forward` method, then
+    perform an AST rewrite to swap out nodes that are not symbolically
+    traceable with a callsite to the FX alternative.
+
+    To support swapping out an AST node, define a new `visit` method on
+    that node. For more details, see:
+    https://docs.python.org/3/library/ast.html#ast.NodeTransformer
+    '''
+
+    def rewrite(self, fn: FunctionType):
+
+        # Get the source string, align everything with the `def`, then
+        # remote unnecessary indentation
```

Comment:
Can you add a before/after example to this comment showing what it does? It took me a little while to understand what exactly was going on

Reply:
BTW this is actually also an open issue for the TorchScript AST-based frontend https://github.com/pytorch/pytorch/issues/25043 so you could also submit a patch there (i.e. frontend.py) if you're so inclined

Reply:
Submitted! When my PR goes through, I can call the `normalize_source_lines` function from here, which will make the code shorter and cleaner.